hack/demo-env/setup-vcluster-env.sh

```diff
@@ -20,29 +30,173 @@ namespace() {
 	IFS=":" s=($1); echo ${s[1]}
 }
 
-trap cleanup EXIT ERR
+trap cleanup EXIT
+trap on_error ERR
+
+# check_for_openshift looks for cluster OpenShift API Resources, and if found, sets OPENSHIFT=true
+check_for_openshift() {
+
+	OPENSHIFT=true
+	if kubectl api-resources | grep -q "openshift.io"; then 
+		OPENSHIFT=
+	fi
```

Comment:
The above logic breaks things on non-OpenShift clusters:
```shell
$ ./hack/demo-env/setup-vcluster-env.sh create
namespace/vcluster-agent-managed created
namespace/vcluster-control-plane created
namespace/vcluster-agent-autonomous created
error: resource mapping not found for name: "anyuid-seccomp-netbind" namespace: "" from "/home/vagrant/src/argocd-agent/hack/demo-env/resources/scc-anyuid-seccomp-netbind.yaml": no matches for kind "SecurityContextConstraints" in version "security.openshift.io/v1"
ensure CRDs are installed first
ERROR: Error occurred, terminating.
Switched to context "default".
Switched to context "default".

```
```suggestion
	OPENSHIFT=
	if kubectl api-resources | grep -q "openshift.io"; then 
		OPENSHIFT=true
	fi
```